balancer/endpointsharding/endpointsharding.go

```diff
@@ -80,18 +80,27 @@ type endpointSharding struct {
 // for endpoints that are no longer present. It also updates all the children,
 // and sends a single synchronous update of the childrens' aggregated state at
 // the end of the UpdateClientConnState operation. If any endpoint has no
-// addresses, returns error without forwarding any updates. Otherwise returns
-// first error found from a child, but fully processes the new update.
+// addresses or there is a duplicate address amongst the endpoints, returns
+// error without forwarding any updates. Otherwise returns first error found
+// from a child, but fully processes the new update.
 func (es *endpointSharding) UpdateClientConnState(state balancer.ClientConnState) error {
 	if len(state.ResolverState.Endpoints) == 0 {
 		return errors.New("endpoints list is empty")
 	}
-	// Check/return early if any endpoints have no addresses.
+
+	addrs := resolver.NewAddressMap()
+	// Check/return early if any endpoints have no addresses, or there is a duplicate address map.
 	// TODO: make this configurable if needed.
 	for i, endpoint := range state.ResolverState.Endpoints {
 		if len(endpoint.Addresses) == 0 {
 			return fmt.Errorf("endpoint %d has empty addresses", i)
 		}
+		for _, addr := range endpoint.Addresses {
+			if _, ok := addrs.Get(addr); ok {
+				return fmt.Errorf("duplicate addr %v present in endpoints list", addr)
```

Comment:
Should this wrap `balancer.ErrBadResolverState` so that the re-resolution is triggered?

Reply:
I don't think so.  This is being fed back to the name resolver anyway.

I don't think this magic error is actually used anymore, and I don't see anything in the channel checking for it.

Reply:
Okay.